File: .sdk-repos/orchestration-cluster-api-js/examples/mapping-rule.ts
// Compilable usage examples for mapping rule operations.
// These examples are type-checked during build to guard against API regressions.

import { createCamundaClient, type MappingRuleId } from '@camunda8/orchestration-cluster-api';

//#region CreateMappingRule
async function createMappingRuleExample(mappingRuleId: MappingRuleId) {
  const camunda = createCamundaClient();

  const result = await camunda.createMappingRule({
    mappingRuleId,
    name: 'LDAP Group Mapping',
    claimName: 'groups',
    claimValue: 'engineering',
  });

  console.log(`Created mapping rule: ${result.mappingRuleId}`);
}
//#endregion CreateMappingRule

//#region GetMappingRule
async function getMappingRuleExample(mappingRuleId: MappingRuleId) {
  const camunda = createCamundaClient();

  const rule = await camunda.getMappingRule(
    { mappingRuleId },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`Rule: ${rule.name} (${rule.claimName}=${rule.claimValue})`);
}
//#endregion GetMappingRule

//#region SearchMappingRule
async function searchMappingRulesExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchMappingRule(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const rule of result.items ?? []) {
    console.log(`${rule.mappingRuleId}: ${rule.name}`);
  }
}
//#endregion SearchMappingRule

//#region UpdateMappingRule
async function updateMappingRuleExample(mappingRuleId: MappingRuleId) {
  const camunda = createCamundaClient();

  await camunda.updateMappingRule({
    mappingRuleId,
    name: 'LDAP Group Mapping',
    claimName: 'groups',
    claimValue: 'engineering-team',
  });
}
//#endregion UpdateMappingRule

//#region DeleteMappingRule
async function deleteMappingRuleExample(mappingRuleId: MappingRuleId) {
  const camunda = createCamundaClient();

  await camunda.deleteMappingRule({ mappingRuleId });
}
//#endregion DeleteMappingRule

// Suppress "declared but never read"
void createMappingRuleExample;
void getMappingRuleExample;
void searchMappingRulesExample;
void updateMappingRuleExample;
void deleteMappingRuleExample;

// Suppress "declared but never read"
void createMappingRuleExample;
void getMappingRuleExample;
void searchMappingRulesExample;
void updateMappingRuleExample;
void deleteMappingRuleExample;

File: .sdk-repos/orchestration-cluster-api-go/examples/mapping_rule.go
// Mapping rule operations: map IdP claims to Camunda identities.
package examples

import (
	"context"
	"fmt"

	camunda "github.com/camunda/orchestration-cluster-api-go"
	openapi "github.com/camunda/orchestration-cluster-api-go/client"
)

func createMappingRuleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region CreateMappingRule
	// Map the IdP claim `groups=auditors` to a Camunda mapping-rule identity.
	result, err := client.CreateMappingRule(ctx,
		*openapi.NewMappingRuleCreateRequest("groups", "auditors", "SSO Auditors", "sso-auditors"))
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion CreateMappingRule
	return nil
}

func searchMappingRuleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchMappingRule
	result, err := client.SearchMappingRule(ctx, *openapi.NewMappingRuleSearchQueryRequest())
	if err != nil {
		return err
	}
	for _, r := range result.GetItems() {
		fmt.Printf("%v\n", r)
	}
	// endregion SearchMappingRule
	return nil
}

func getMappingRuleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetMappingRule
	rule, err := client.GetMappingRule(ctx, "sso-auditors")
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", rule)
	// endregion GetMappingRule
	return nil
}

func updateMappingRuleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region UpdateMappingRule
	result, err := client.UpdateMappingRule(ctx, "sso-auditors",
		*openapi.NewMappingRuleUpdateRequest("groups", "senior-auditors", "SSO Senior Auditors"))
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion UpdateMappingRule
	return nil
}

func deleteMappingRuleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region DeleteMappingRule
	return client.DeleteMappingRule(ctx, "sso-auditors")
	// endregion DeleteMappingRule
}
